common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -828,6 +916,31 @@ private void lazyExecute0(@Schedule Runnable task) {
         execute(ObjectUtil.checkNotNull(task, "task"), false);
     }
 
+    @Override
+    void scheduleRemoveScheduled(final ScheduledFutureTask<?> task) {
+        ObjectUtil.checkNotNull(task, "task");
+        int currentState = state;
+        if (supportSuspension && currentState == ST_SUSPENDED) {
+            // In the case of scheduling for removal we need to also ensure we will recover the "suspend" state
+            // after it if it was set before. Otherwise we will always end up "unsuspending" things on cancellation
+            // which is not optimal.
+            execute(new Runnable() {
+                @Override
+                public void run() {
+                    task.run();
+                    if (canSuspend(currentState)) {
```

Comment:
The `currentState` is always `ST_SUSPENDED` here. Maybe extract a separate check-method out of `canSuspend`?

Reply:
This would mess up when sub-classes override `canSuspend` which is true for our event loop impls